mantidimaging/gui/windows/stack_properties_dialog/presenter.py

```diff
@@ -0,0 +1,36 @@
+# Copyright (C) 2021 ISIS Rutherford Appleton Laboratory UKRI
+# SPDX - License - Identifier: GPL-3.0-or-later
+from __future__ import annotations
+
+from pathlib import Path
+from typing import TYPE_CHECKING
+
+from mantidimaging.core.utility.size_calculator import full_size_MB
+from mantidimaging.gui.mvp_base import BasePresenter
+
+if TYPE_CHECKING:
+    from mantidimaging.gui.windows.stack_properties_dialog.view import StackPropertiesDialog
+
+
+class StackPropertiesPresenter(BasePresenter):
+
+    def __init__(self, view: StackPropertiesDialog):
+        super().__init__(view)
+
+    def set_stack_data(self) -> None:
+        if self.view.stack is not None:
```

Comment:
Not sure if this is from a mypy suggestion or older version of the code.
If `self.view.stack` is `None` then something has gone wrong (e.g. side bar has got out of sync), so I don't think it should be silently ignored here.

It might be better handled with an `assert self.stack is not None` in the view, so that the rest of the dialog can assume that it is set. Or this and the call to `get_images_by_uuid()` could move into the main window.